superset/db_engine_specs/databend.py

```diff
@@ -0,0 +1,353 @@
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+import re
+from datetime import datetime
+from typing import Any, cast, TYPE_CHECKING
+
+from flask_babel import gettext as __
+from marshmallow import fields, Schema
+from marshmallow.validate import Range
+from sqlalchemy import types
+from sqlalchemy.engine.url import URL
+from urllib3.exceptions import NewConnectionError
+
+from superset.databases.utils import make_url_safe
+from superset.db_engine_specs.base import (
+    BaseEngineSpec,
+    BasicParametersMixin,
+    BasicParametersType,
+    BasicPropertiesType,
+)
+from superset.db_engine_specs.exceptions import SupersetDBAPIDatabaseError
+from superset.errors import ErrorLevel, SupersetError, SupersetErrorType
+from superset.utils.core import GenericDataType
+from superset.utils.hashing import md5_sha_from_str
+from superset.utils.network import is_hostname_valid, is_port_open
+
+if TYPE_CHECKING:
+    from superset.models.core import Database
+
+logger = logging.getLogger(__name__)
+
+
+class DatabendBaseEngineSpec(BaseEngineSpec):
+    """Shared engine spec for Databend."""
+
+    time_secondary_columns = True
+    time_groupby_inline = True
+
+    _time_grain_expressions = {
+        None: "{col}",
+        "PT1M": "to_start_of_minute(TO_DATETIME({col}))",
```

Comment:
Could you use `superset.constants.TimeGrain`?

`TimeGrain.MINUTE: "to_start_of_minute(TO_DATETIME({col}))",`